polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -778,14 +998,23 @@ void SYCLRaiseHostConstructsPass::runOnOperation() {
   MLIRContext *context = &getContext();
 
   RewritePatternSet rewritePatterns{context};
-  rewritePatterns.add<RaiseKernelName, BufferInvokeConstructorPattern,
-                      AccessorInvokeConstructorPattern, RaiseIDConstructor,
-                      RaiseRangeConstructor>(context);
+  rewritePatterns
+      .add<BufferInvokeConstructorPattern, AccessorInvokeConstructorPattern>(
+          context);
 
   // RaiseKernelName should be prioritized, as RaiseSetKernel depends on that.
   rewritePatterns.add<RaiseKernelName>(context, /*benefit=*/2);
   rewritePatterns.add<RaiseSetKernel>(context, /*benefit=*/1);
 
+  // Raising of some constructors (id, range and nd_range) should be
+  // prioritized, as RaiseSetNDRange depends on those. Also, raising of id and
+  // range constructors should be prioritized, as nd_range constructor uses
+  // them.
+  rewritePatterns.add<RaiseIDConstructor, RaiseRangeConstructor>(context,
```

Comment:
[nit]: Can you reorder the patterns to put the one with higher benefits "higher" ?

Reply:
I think I'd rather stick to the logical order we have now:
1. buffer + accessor construction\
2. kernel name setting
3. nd-range related constructors
4. set-nd-range

Reply:
Note the "per benefit" ordering holds in each category